src/main/java/org/folio/fqm/service/EntityTypeService.java

```diff
@@ -112,34 +107,24 @@ public ColumnValues getFieldValues(UUID entityTypeId, String fieldName, @Nullabl
       .findFieldDefinition(new FqlField(fieldName), entityType)
       .orElseThrow(() -> new FieldNotFoundException(entityType.getName(), fieldName));
 
-    if (field.getName().equals("tenant_id")) {
-      // Hackish way to get tenant values for cross-tenant instance queries. Can be removed after completion of MODFQMMGR-335.
-      List<String> tenants = crossTenantQueryService.getTenantsToQuery(UUID.fromString("6b08439b-4f8e-4468-8046-ea620f5cfb74"));
-      List<ValueWithLabel> tenantValues = tenants
-        .stream()
-        .map(tenant -> new ValueWithLabel().value(tenant).label(tenant))
-        .toList();
-      return new ColumnValues().content(tenantValues);
-    }
-
-    if (field.getName().equals("source_tenant_id")) {
-      // Hackish way to get tenant values for cross-tenant instance queries. Can be removed after completion of MODFQMMGR-335.
-      List<String> tenants = crossTenantQueryService.getTenantsToQuery(UUID.fromString("6b08439b-4f8e-4468-8046-ea620f5cfb74"));
-      if (tenants.size() > 1) {
-        List<ValueWithLabel> tenantValues = tenants
-          .stream()
-          .map(tenant -> new ValueWithLabel().value(tenant).label(tenant))
-          .toList();
-        return new ColumnValues().content(tenantValues);
-      } else {
-        // tenant is not central tenant, but we need to provide central tenant id as an available value
-        String centralTenantId = crossTenantQueryService.getCentralTenantId();
-        List<ValueWithLabel> tenantList = new ArrayList<>();
-        if (centralTenantId != null && !centralTenantId.equals(tenants.get(0))) {
-          tenantList.add(new ValueWithLabel().value(centralTenantId).label(centralTenantId));
+    if (field.getSource() != null) {
+      if (field.getSource().getType() == SourceColumn.TypeEnum.ENTITY_TYPE) {
+        return getFieldValuesFromEntityType(entityType, fieldName, searchText);
+      } else if (field.getSource().getType() == SourceColumn.TypeEnum.FQM) {
+        switch (Objects.requireNonNull(field.getSource().getName(), "Value sources with the FQM type require the source name to be configured")) {
+          case "currency" -> {
```

Comment:
Before, currencies were handled by looking for a column named `pol_currency`, which doesn't work since we renamed that column. Now, since it uses the `name` from `source`, the column name is irrelevant, so went with `currency`